R/testdata.R

```diff
@@ -695,12 +695,42 @@ dataset_test_worker <-
           expect_true(all(i),
                       info = paste0(
                         f,
-                        "-site names from metadata not present in data file: ",
+                        "- site names from metadata not present in data file: ",
                         names(metadata$locations)[!i]
                       ))
         }
 
-
+        ## Check that dataset can pivot wider
+        expect_no_error(dataset <- test_build_dataset(
+          file.path(path_data, dataset_id, "metadata.yml"),
+          file.path(path_data, dataset_id, "data.csv"),
+          dataset_id,
+          get_schema("config/traits.yml", "traits"),
+          get_unit_conversions("config/unit_conversions.csv"),
+          get_schema(),
+          get_schema("config/metadata.yml",  "metadata"),
+          read_csv_char("config/taxon_list.csv")
+        ), info = sprintf(" - cannot build %s", dataset_id))
+
+        expect_equal(
+          dataset$traits %>%
+          select(
+            dplyr::all_of(c("dataset_id", "trait_name", "value", "observation_id", "source_id", "taxon_name",
+            "entity_type", "life_stage", "basis_of_record", "value_type", "population_id", "individual_id",
+            "temporal_id", "method_id", "method_context_id", "entity_context_id", "original_name"))
+          ) %>%
+          pivot_wider(names_from = "trait_name", values_from = "value", values_fn = length) %>%
+          pivot_longer(cols = 16:ncol(.)) %>%
```

Comment:
Where does the 16 come from? If from the vector of names above, I suggest adding that as a vector above, then drawing the number from the length of the vector. 

Hard coded numbers are dangerous, as they don't adjust when we change something

Reply:
After pivoting wider column 16 marks the beginning of the columns with traits, while columns 1-15 are the variables above (minus trait_name which becomes the column headers and value). The number of columns with traits is variable dataset by dataset, but as long as you pivot across those variables, they always start at column 16.